src/dataset.jl

```diff
@@ -172,16 +172,133 @@ end
     write(dataset::AbstractDataset, filename::AbstractString; kwargs...)
 
 Writes the dataset to the designated filename.
+
+For the keyword arguments for vector datasets, consult the documentation of `writelayers` 
+and for raster datasets the docs for `writerasters`.
 """
 function write(
     dataset::AbstractDataset,
     filename::AbstractString;
     kwargs...,
 )::Nothing
+    
+    if nraster(dataset) > 0
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
```

Comment:
I don't know of any drivers that support both, raster data + `createlayer`, so this should never be the case, correct?

Reply:
I'm not sure if this is what you're asking, but there are several driver names that occur in both https://gdal.org/drivers/raster/index.html and https://gdal.org/drivers/vector/index.html, like `netCDF`. Though that driver is not yet included in the build.

Reply:
Yes, right, but I've never seen a gdal dataset containing raster and vector data. But that's maybe my limited experience. Is there a driver included in the build which supports both? Wouldn't it be good to have netCDF support? And if there are drivers which support both, how could a write function look like? Or should we just skip that case for the time being?

Reply:
Yeah I hope we can soon add netCDF support to the build. I haven't seen datasets containing both either so far, so if it is difficult we could leave it for later.

Reply:
Okay, since we don't yet have a dataset with both rasters and vectors (or know if we can create it), I'm okay with resolving this thread in lieu of https://github.com/yeesian/ArchGDAL.jl/pull/315/files#r943079678. Thank you so much for the discussion!